pkg/peers/peers.go

```diff
@@ -111,37 +111,49 @@ func (p *Peers) updatePeers(ctx context.Context, getSelector func() labels.Selec
 	if err := p.List(readerCtx, &nodes, client.MatchingLabelsSelector{Selector: getSelector()}); err != nil {
 		if errors.IsNotFound(err) {
 			// we are the only node at the moment... reset peerList
-			p.workerPeersAddresses = [][]v1.NodeAddress{}
+			p.workerPeersAddresses = []string{}
 		}
 		p.log.Error(err, "failed to update peer list")
 		return
 	}
 
+	pods := v1.PodList{}
+	listOptions := &client.ListOptions{
+		LabelSelector: labels.SelectorFromSet(labels.Set{
+			"app.kubernetes.io/name":      "self-node-remediation",
+			"app.kubernetes.io/component": "agent",
+		}),
+	}
+	if err := p.List(readerCtx, &pods, listOptions); err != nil {
+		p.log.Error(err, "could not get pods")
+	}
+
 	nodesCount := len(nodes.Items)
-	addresses := make([][]v1.NodeAddress, nodesCount)
+	addresses := make([]string, nodesCount)
 	for i, node := range nodes.Items {
-		addresses[i] = node.Status.Addresses
+		for _, pod := range pods.Items {
+			if pod.Spec.NodeName == node.Name {
+				addresses[i] = pod.Status.PodIP
```

Comment:
I was wondering why we have a string type now, and indeed there is a better choice IMHO, what about using `pod.Status.PodIPs[0]`?

Reply:
Do you mean moving around this data as [PodIP](https://github.com/medik8s/self-node-remediation/blob/c6a8cd32b8191d88020fd637cc9ca23de69b1078/vendor/k8s.io/api/core/v1/types.go#L3815-L3818) and then let [popPeersIP](https://github.com/medik8s/self-node-remediation/blob/c6a8cd32b8191d88020fd637cc9ca23de69b1078/pkg/apicheck/check.go#L216) deal with it returning the `string[]` of IPs?

Reply:
I mean, at which point of the chain would it be better to use the underlining `PodIP.IP`?
IIUC, the only interface requiring the `string` is [grpc.DialContext](https://pkg.go.dev/google.golang.org/grpc@v1.56.3#DialContext)

Reply:
I would use PodIP everywhere where we used NodeAddress before.
But: oh, it's just a wrapper around a string, I expected a more IP-ish thing 😁  
And: oh, we did not even check the type of the NodeAddress in the old version of popNodes, and just took the first one 🙈

Reply:
>  But: oh, it's just a wrapper around a string, I expected a more IP-ish thing 😁

:grin: yep, moreover at the end of the day we use the `string`, so not sure it's worth it